ksqldb-rest-client/src/main/java/io/confluent/ksql/rest/client/KsqlTarget.java

```diff
@@ -451,8 +451,10 @@ private KsqlRequest createKsqlRequest(
         if (begin != i) { // Ignore random newlines - the server can send these
           final Buffer sliced = buff.slice(begin, i);
           final Buffer tidied = StreamPublisher.toJsonMsg(sliced, true);
-          final StreamedRow row = deserialize(tidied, StreamedRow.class);
-          rows.add(row);
+          if (tidied.length() > 0) {
+            final StreamedRow row = deserialize(tidied, StreamedRow.class);
+            rows.add(row);
+          }
```

Comment:
I happened to find a bug during the implementation. `toJsonMsg` in the prior line may trim everything out and return an empty string, which we can't deserialize, so we shouldn't try.